src/main/java/org/codehaus/mojo/license/api/DependenciesTool.java

```diff
@@ -163,8 +171,25 @@
 
                 try
                 {
-                    depMavenProject =
-                        mavenProjectBuilder.buildFromRepository( artifact, remoteRepositories, localRepository, true );
+                    /*
+                     * Some POMs may refer to e.g. java.home or java.version,
+                     * so we need to ensure common system properties are set when parsing POMs.
+                     * Using the current system properties seems like a decent solution.
+                     */
+                    Properties props = new Properties();
+                    props.putAll( System.getProperties() );
+                    ProjectBuildingRequest req = new DefaultProjectBuildingRequest()
```

Comment:
Do we need a new instance of ProjectBuildingRequest for each iteration of the enclosing for loop?

Reply:
The parameters don't change, so probably not. I'll change it to reuse the request.

Reply:
I think it is better here. Constructing the request shouldn't take any real time, and moving it outside the for loop puts a lot of distance between the declaration and use of the request.

Reply:
> Constructing the request shouldn't take any real time

It very much depends on how many iterations the loop has and also on how many sys props are set.

> puts a lot of distance

Sounds like a readability/aesthetics concern? - those are important, but here, I'd favor performance, unless proven that pulling the req out of the loop is technically impossible.

Reply:
Sure, but the system properties aren't being copied anymore, since the session change.

It's just a readability concern.  

I'll move it out of the loop.

Reply:
Done